tcmur_aio.c

```diff
@@ -78,6 +79,26 @@ void track_aio_request_finish(struct tcmur_device *rdev, int *is_idle)
 	pthread_cleanup_pop(0);
 }
 
+void track_aio_wakeup_finish(struct tcmur_device *rdev, int *wake_up)
+{
+	struct tcmu_track_aio *aio_track = &rdev->track_queue;
+
+	pthread_cleanup_push(_cleanup_mutex_lock, (void *)&aio_track->track_lock);
```

Comment:
I don't think cleanup is necessary here.  It's necessary when there is a cancellation point in the critical section.  My reading of posix (and the linux man page) doesn't include assert as a cancellation point, but I could be wrong...  It may output, which could block (and hence should count as a cancellation point), but if it does, it will exit the application anyway so is probably moot.

Reply:
Yeah, it's probably not needed since the daemon will crash if we hit the assert.

I think we are sometimes overly careful. We can do a clean up patch later.